ANNOTATION.md

```diff
@@ -1,142 +1,142 @@
-| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Modifier'\]/self::\*\[@Token='private'\] | VisibleFromInstance |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='QualifiedName'\] | Expression, Identifier, Qualified |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='SimpleName'\] | Expression, Identifier |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[\(@InternalType='MethodDeclaration'\) or \(@InternalType='TypeDeclaration'\)\]/self::\*\[child::\(@InternalType='Modifier'\) and \(@Token='public'\)\] | Visibility, World |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[\(@InternalType='MethodDeclaration'\) or \(@InternalType='TypeDeclaration'\)\]/self::\*\[child::\(@InternalType='Modifier'\) and \(@Token='private'\)\] | Visibility, Type |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[\(@InternalType='MethodDeclaration'\) or \(@InternalType='TypeDeclaration'\)\]/self::\*\[child::\(@InternalType='Modifier'\) and \(@Token='protected'\)\] | Visibility, Subtype |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[\(@InternalType='MethodDeclaration'\) or \(@InternalType='TypeDeclaration'\)\]/self::\*\[not\(child::\(@InternalType='Modifier'\) and \(\(@Token='public'\) or \(@Token='private'\) or \(@Token='protected'\)\)\)\] | Visibility, Package |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='PackageDeclaration'\] | Declaration, Package |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='ImportDeclaration'\] | Declaration, Import |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='ImportDeclaration'\]/\*\[@InternalType='QualifiedName'\] | Pathname, Import |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='AnonymousClassDeclaration'\] | Expression, Declaration, Type, Incomplete |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='AnonymousClassDeclaration'\]/\*\[@internalRole\]\[@internalRole='bodyDeclarations'\] | Body |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='AnnotationTypeDeclaration'\] | Declaration, Type, Incomplete |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='AnnotationTypeDeclaration'\]/\*\[@internalRole\]\[@internalRole='bodyDeclarations'\] | Body |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='EnumDeclaration'\] | Declaration, Type, Incomplete |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='TypeDeclaration'\] | Declaration, Type |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='TypeDeclarationStatement'\] | Statement, Declaration, Type, Incomplete |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='MethodDeclaration'\] | Declaration, Function |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='MethodDeclaration'\]/\*\[@internalRole\]\[@internalRole='name'\] | Function, Name |
```

Comment:
These shouldn't have also `Declaration`?

Reply:
Good question. I'm not sure about it, but I've considered that the `Declaration` role could be part of the parent node, while its children have different roles, in the same way that, in the case of the `If` constructs, there's a parent `If`, `Statement`, while the children don't have the `Statement` role anymore, even if they're part of the if statement (which is already noted using the `If` role).
This makes it easier to differentiate the declaration node itself from its nodes, and that's why I think it may be better this way, but it's arguable.

Reply:
Yes, this is something I also tough about, specially with some monster role lists for some nodes where I apply the Declaration and similar roles, we should probably decide what to do in this case and document it somewhere.